plugin/core/compat.gradle

```diff
@@ -0,0 +1,56 @@
+import org.apache.tools.ant.taskdefs.condition.Os
+
+String gw = Os.isFamily(Os.FAMILY_WINDOWS) ? 'gradlew.bat' : './gradlew'
+String compatClassesDestination = 'build/compat/classes'
+TaskContainer taskContainer = project.tasks
+
+File[] compatModulesDirs = rootProject.file('plugin/compat').listFiles(new FilenameFilter() {
+
+    /**
+     * We want to collect all the folders inside plugin/compat that are also standalone Gradle projects,
+     * hence we look for their settings.gradle
+     */
+    @Override
+    boolean accept(File dir, String name) {
+        return new File(dir, "$name/settings.gradle").exists()
+    }
+})
+
+Task compatClassesTask = taskContainer.create(name: 'releasePluginCompatClasses', type: DefaultTask) {
```

Comment:
This task is used to 'join' all the work needed to compile and include the compat classes in the `plugin/core` module

Reply:
What's the minimum version of Gradle you need to support?
You may want to consider using `tasks.register` if you can use the 4.8+ Gradle API.

Reply:
Hi @JLLeitschuh thanks for the comment. These tasks here are not part of the plugin code, but just of its builscript, so we could use a more recent version of Gradle and use the new tasks api, as right now the plugin core module is compiled against Gradle 4.3. I will think about it, but it's definitely something that I wouldn't address as part of this tranche of work (that focuses on fixing the compatibility issue of its internals).